docs/general/kakarot_components.md

```diff
@@ -29,43 +31,55 @@ contracts is stored as follows:
 transparent proxy.
 
-## Contract Accounts
+## Accounts
 
-A _Contract Account_ is a StarkNet contract. However, it also acts as an
-Ethereum style contract account within the Kakarot EVM. In isolation it is not
-more than a StarkNet contract that stores some bytecode as well as some
-key-value pairs which were assigned to it on creation. It is only addressable
-via its StarkNet address and not an EVM address (which it is associated with
-inside the Kakarot EVM).
+The Ethereum protocol defines two kind of accounts:
 
-## Externally Owned Account
+- Externally Owned Accounts (EOA): accounts managed by a private-key and a
+  wallet
+- Contract Accounts (CA): accounts managed by their stored code (smart
+  contracts)
 
-Each Externally Owned Account in the EVM world has its counterpart in Starknet
-by the mean of a specific account contract deployed by Kakarot.
+Each of these account has a given Starknet contract class counterpart in
+Kakarot.
 
-This contract is a regular account contract in the Starknet sense with
-`__validate__` and `__execute__` entrypoint. However, it does decode and
-validate an EVM signed transaction and redirect it only to Kakarot. Further
-development will allow the user to have one single Starknet account for both
-Starknet native and Kakarot deployed dApp. For a general introduction to EVM
-transactions, see
-[the official doc](https://ethereum.org/en/developers/docs/transactions/).
+In the EVM, each account (EOA or CA) is located at an address and has the
+following properties:
+
+```json
+{
+  "balance": "0x0ba1a9ce0ba1a9ce",
+  "code": "0x6064640fffffffff2060005500",
+  "nonce": "0x00",
+  "storage": {
+    "0x01": "0x02"
+  }
+}
+```
 
-## Blockhash Registry
+Currently, Kakarot uses an external ERC0 contract for storing the balances and
+consequently only the three other fields are stored in the corresponding
+Starknet contracts.
 
-The [BLOCKHASH](https://www.evm.codes/#40) opcode is a particular opcode that
-requires the EVM to be aware of past blocks
-([see also](https://ethresear.ch/t/the-curious-case-of-blockhash-and-stateless-ethereum/7304/7)).
-Since this is not feasible from within Starknet, we deployed a block hash
-registry contract on Starknet to make this data accessible on-chain.
+### Contract Accounts
 
-The blockhash registry enables this by holding a `block_number -> block_hash`
-mapping that admins can write to and Kakarot core can read from.
+It is basically used only as a storage backend for an EVM contract account. More
+precisely, it uses regular Starknet `@storage_var` to store both the contract
+bytecode and the contract storage (`SSTORE` and `SLOAD`).
+
+### Externally Owned Account
+
+This contract is an account in the Starknet sense, meaning that it defines the
+`__validate__` and `__execute__` entrypoint and is used to send transactions
+from a wallet to Kakarot. However, it only uses the `calldata` field of a
+Starknet transaction, where the raw decoded signed Ethereum transaction bytecode
```

Comment:
I think in the calldata of the Starknet transaction, we have the unsigned encoded Ethereum transaction bytecode. Signature is extracted and injected into the signature field of the Starknet transaction. And decode step is done in the `__validate__` and `__execute__` entrypoints